guesser.rb

```diff
@@ -0,0 +1,62 @@
+#global vars
+is_game_active = true
+is_new_game = true
+play_again = nil
+lower_bound = 1
+upper_bound = 100
+
+while is_game_active
+
+  if is_new_game
+    puts "Enter a number between 1 and 100"
+    random_num = gets.chomp
+    lower_bound = 1
+    upper_bound = 100
+    play_again = nil
+
+    until random_num.to_i >= 1 && random_num.to_i <= 100
+      puts('Invalid input. Enter a number between 1 and 100!')
+      random_num = gets.chomp
+    end
+    is_new_game = false
+  end
+
+
+  machine_guess = (upper_bound + lower_bound) / 2
+
+  puts "My guess is #{machine_guess}. Is my guess correct, too high or too low? correct/too high/too low"
+
+  response = gets.chomp
+  until response == "correct" or response == "too low" or response == "too high" do
+    response = gets.chomp
```

Comment:
❕ Might be nice to show the message to the user again if they keep inputting invalid responses.